serverjs/cubefn.js

```diff
@@ -228,6 +252,172 @@ async function getElo(cardnames, round) {
   return result;
 }
 
+function CSVtoCards(cards, carddb) {
+  let missing = '';
+  const newCards = [];
+  const newMaybe = [];
+  for (const rawCard of cards) {
+    const split = util.CSVtoArray(rawCard);
+    const name = split[0];
+    const maybeboard = split[8];
+    const card = {
+      name,
+      cmc: split[1],
+      type_line: split[2].replace('-', '—'),
+      colors: split[3].split('').filter((c) => [...'WUBRG'].includes(c)),
+      set: split[4].toUpperCase(),
+      addedTmsp: new Date(),
+      collector_number: split[5],
+      status: split[6],
+      finish: split[7],
+      imgUrl: split[9] && split[9] !== 'undefined' ? split[9] : null,
+      tags: split[10] && split[10].length > 0 ? split[10].split(',') : [],
+    };
+
+    const potentialIds = carddb.allIds(card);
+    if (potentialIds && potentialIds.length > 0) {
+      // First, try to find the correct set.
+      const matchingSetAndNumber = potentialIds.find((id) => {
+        const dbCard = carddb.cardFromId(id);
+        return (
+          card.set.toUpperCase() === dbCard.set.toUpperCase() &&
+          card.collector_number.toUpperCase() === dbCard.collector_number.toUpperCase()
+        );
+      });
+      const matchingSet = potentialIds.find((id) => carddb.cardFromId(id).set.toUpperCase() === card.set);
+      const nonPromo = potentialIds.find(carddb.notPromoOrDigitalId);
+      const first = potentialIds[0];
+      card.cardID = matchingSetAndNumber || matchingSet || nonPromo || first;
+      if (maybeboard === 'true') {
+        newMaybe.push(card);
+      } else {
+        newCards.push(card);
+      }
+    } else {
+      missing += `${card.name}\n`;
+    }
+  }
+  return { newCards, newMaybe, missing };
+}
+
+async function generateBlogpost(req, res, cube, changelog, added, missing, carddb) {
```

Comment:
Can we name this function something more appropriate and put it back into cube_routes? It's doing stuff other than generating the blogpost and I don't like having what is effectively a route handler outside the routes files.